File: bench/Benchmark.py
from iron_mq import *

testClient = IronMQ(host="mq-aws-us-east-1.iron.io",
    project_id="54a241faaeffe60005000047",
    token="DG0HDEUlDvj-oyTdM9Ua9rTxw7w",
    protocol="https", port=443,
    api_version=1,
    config_file=None)

def doStuff():
    queues = testClient.queues
    queue = testClient.queue("default")
    queue.post("This is message number ")
    messageList = queue.get()
    messageId = messageList['messages'][0]['id']
    queue.delete(messageId)

if __name__ == "__main__":
    doStuff()

# if __name__ == "__main__":
#     import timeit
#     print timeit.timeit("doStuff()", setup="from __main__ import doStuff", number = 100)

File: bench/Benchmark.hs
{-# LANGUAGE OverloadedStrings #-}

import Criterion.Main

import Network.IronMQ
import Network.IronMQ.Types

main :: IO ()
main = defaultMain [bench "get queue info, post a message, get messages, delete message" $ nfIO (doStuff)]

testClient :: Client
testClient = Client {
    token = "DG0HDEUlDvj-oyTdM9Ua9rTxw7w",
    projectID = "54a241faaeffe60005000047",
    server = "mq-aws-us-east-1.iron.io",
    apiVersion = "1"
}


doStuff :: IO ()
doStuff = do
    _ <- queues testClient
    postMessages testClient "default" [message{mBody = "This is message number "}]
    messageList <- getMessages testClient "default"
    let messageID = mId (head (mlMessages messageList))
    case messageID of
        Just x -> deleteMessage testClient "default" x
    return ()